File: app/models/address.rb
class Address
  include Mongoid::Document
  
  field :name, :type => String
  field :city, :type => String
  field :country, :type => String, :default => 'US'
  field :first_line, :type => String
  field :second_line, :type => String
  field :state, :type => String
  field :zip, :type => String
  
  embedded_in :customer
  embedded_in :store
  
  liquid_methods :name, :city, :country, :first_line, :second_line, :state, :zip
end

File: app/assets/javascripts/pos/models/address.js
SwapzPOS.Address = Ember.Object.extend({
  id: null,
  city: null,
  country: null,
  firstLine: null,
  name: null,
  secondLine: null,
  state: null,
  type: null,
  zip: null,
  _remove: false,
  formatted: function() {
    return this.get('firstLine') + ' ' + this.get('secondLine') + ' ' + this.get('city') + ' ' + this.get('state') + ', ' + this.get('zip') + ' ' + this.get('country');
  }.property('city', 'country', 'firstLine', 'secondLine', 'state', 'type', 'zip')
});